contracts/workflows/RoyaltyWorkflows.sol

```diff
@@ -42,208 +42,130 @@ contract RoyaltyWorkflows is IRoyaltyWorkflows, MulticallUpgradeable, AccessMana
-
-        // Claims revenue for each specified currency token from the latest snapshot
+        // Claims revenue for each specified currency token
         amountsClaimed = ancestorIpRoyaltyVault.claimRevenueOnBehalfByTokenBatch({
-            snapshotId: snapshotId,
-            tokenList: _getCurrencyTokenList(royaltyClaimDetails),
-            claimer: claimer
+            claimer: claimer,
+            tokenList: _getUniqueCurrencyTokens(currencyTokens)
         });
     }
 
-    /// @notice Transfers royalties to the ancestor IP's royalty vault, takes a snapshot, claims revenue for each
-    /// specified currency token both on the new snapshot and on each specified unclaimed snapshots.
-    /// @param ancestorIpId The address of the ancestor IP.
-    /// @param claimer The address of the claimer of the revenue tokens (must be a royalty token holder).
-    /// @param unclaimedSnapshotIds The IDs of unclaimed snapshots to include in the claim.
-    /// @param royaltyClaimDetails The details of the royalty claim from child IPs,
-    /// see {IRoyaltyWorkflows-RoyaltyClaimDetails}.
-    /// @return snapshotId The ID of the snapshot taken.
-    /// @return amountsClaimed The amounts of revenue claimed for each currency token.
-    function transferToVaultAndSnapshotAndClaimBySnapshotBatch(
+    /// @notice Transfers all avaiable royalties from various royalty policies to the royalty
+    ///         vault of an ancestor IP, and claims all the revenue for each currency token
+    ///         from the ancestor IP's royalty vault to the claimer.
+    /// @param ancestorIpId The address of the ancestor IP from which the revenue is being claimed.
+    /// @param claimer The address of the claimer of the currency (revenue) tokens.
+    /// @param childIpIds The addresses of the child IPs from which royalties are derived.
+    /// @param royaltyPolicies The addresses of the royalty policies that govern royalty flows for each child IP.
+    /// @param currencyTokens The addresses of the currency tokens in which the royalties will be claimed.
+    /// @return amountsClaimed The amounts of successfully claimed revenue for each specified currency token.
+    function claimAllRevenue(
         address ancestorIpId,
         address claimer,
-        uint256[] calldata unclaimedSnapshotIds,
-        RoyaltyClaimDetails[] calldata royaltyClaimDetails
-    ) external returns (uint256 snapshotId, uint256[] memory amountsClaimed) {
-        // Transfers to ancestor's vault an amount of revenue tokens claimable via the given royalty policy
-        for (uint256 i = 0; i < royaltyClaimDetails.length; i++) {
-            IGraphAwareRoyaltyPolicy(royaltyClaimDetails[i].royaltyPolicy).transferToVault({
-                ipId: royaltyClaimDetails[i].childIpId,
+        address[] calldata childIpIds,
+        address[] calldata royaltyPolicies,
+        address[] calldata currencyTokens
+    ) external returns (uint256[] memory amountsClaimed) {
```

Comment:
Looks like the function is for claim revenue from each child, in this relationship `childIpId -> (royaltyPolicy, currencyToken)`. If so, the function comments would be better to enhance to clarify.

Reply:
Updated comments in [da5bcc9](https://github.com/storyprotocol/protocol-periphery-v1/pull/118/commits/da5bcc9996e4857ff154575d0094ea1d557d45e2).